### PR TITLE
Creation of the React App + Testing Environments + README update for setup

### PR DESCRIPTION
**Pull request by lyubageorgieva**
This task pertains to the creation of the React app using React + TypeScript + Vite. All the steps for the setup have been documented in the README file. Additionally, in this PR, we have the setup of unit tests using Vitest, the setup of component tests using Cypress, and the setup of E2E tests using Cypress. Eslint was set up for proper code quality. Lastly, there is a coverage report that was set up for the unit tests using Vitest.

This task is from the US - Root Menu Front End [#11 ]

### CODE REVIEW

**Comment by MariaR001** [comment]
Looks good! works as expected

**Review by MariaR001 — APPROVED** [review]
works as expected!

**Review by WillTrem — CHANGES_REQUESTED** [review]
Looks super good in general! Good job! Just that one typo in the README to fix, nothing crazy

File: README.md [hunk 1]
```diff
@@ -34,4 +34,133 @@ You can stop them at any time by running `docker compose stop`.
...
+
+To build component tests with Cypress Component Testing
+
+```bash
+npm run cy:open-component
+```
+
+### Integration Tests 
+
+The [integration tests](https://docs.cypress.io/guides/component-testing/react/overview#React-with-Vite) will be handled by Cypress. These tests will be generated in the `cypress/component` directory.
```
[the remark this answers is not in the document]
Comment (lyubageorgieva) [reply README.md]: Good catch! I'll fix it.

**Review by WillTrem — APPROVED** [review]
Super!

**Pull request merged**